src/functionalTest/java/uk/gov/hmcts/reform/waworkflowapi/controllers/EvaluateDmnTest.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.reform.waworkflowapi.controllers;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.waworkflowapi.SpringBootFunctionalBaseTest;
+import uk.gov.hmcts.reform.waworkflowapi.external.taskservice.DmnValue;
+import uk.gov.hmcts.reform.waworkflowapi.external.taskservice.EvaluateDmnRequest;
+import uk.gov.hmcts.reform.waworkflowapi.utils.AuthorizationHeadersProvider;
+
+import java.util.Map;
+
+import static net.serenitybdd.rest.SerenityRest.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.reform.waworkflowapi.config.ServiceTokenGeneratorConfiguration.SERVICE_AUTHORIZATION;
+
+public class EvaluateDmnTest extends SpringBootFunctionalBaseTest {
+
+    @Autowired
+    public AuthorizationHeadersProvider authorizationHeadersProvider;
+
+    private String serviceAuthorizationToken;
+
+    @Before
+    public void setUp() {
+        serviceAuthorizationToken =
+            authorizationHeadersProvider
+                .getAuthorizationHeaders()
+                .getValue(SERVICE_AUTHORIZATION);
+    }
+
+    @Test
+    public void should_not_allow_requests_without_valid_service_authorisation_and_return_403_response_code() {
+
+        given()
+            .relaxedHTTPSValidation()
+            .contentType(APPLICATION_JSON_VALUE)
+            .body(new EvaluateDmnRequest(null)).log().body()
+            .baseUri(testUrl)
+            .pathParam("key", WA_TASK_INITIATION_IA_ASYLUM)
+            .basePath("/workflow/decision-definition/{key}/evaluate")
+            .when()
+            .post()
+            .then()
+            .statusCode(HttpStatus.FORBIDDEN_403);
+    }
+
+    @Test
+    public void should_evaluate_and_return_dmn_results() {
+        given()
+            .relaxedHTTPSValidation()
+            .header(SERVICE_AUTHORIZATION, serviceAuthorizationToken)
+            .contentType(APPLICATION_JSON_VALUE)
+            .body(new EvaluateDmnRequest(mockVariables())).log().body()
+            .baseUri(testUrl)
+            .pathParam("key",WA_TASK_INITIATION_IA_ASYLUM)
+            .basePath("/workflow/decision-definition/key/{key}/evaluate")
+            .when()
+            .post()
+            .prettyPeek()
```

Comment:
I think we should not output to the console

Reply:
ok